### DISCUSSION TITLE
Using Another Video File as Background Fallback Option for Background Video

### DISCUSSION THREAD

**Discussion by xaelisart** [elementor/elementor#24904] [Feature Request]
### Prerequisites

- [X] I have searched for similar feature requests in both open and closed discussions and cannot find an existing request.
- [X] I have verified that the feature is still missing from the latest stable versions of Elementor or Elementor Pro.

### What problem is your feature request going to solve? Please describe..

Hello,

iOS and Safari browsers currently does not fully support for WEBM video. I spoke with support team and the solution was to simply not use WEBM video. Since we already have "Background Fallback" option for images i would like to ask for a "Background Fallback Video" option. Similar to the multiple video sources used in html.

example;

```
<video autoplay muted playsinline loop>
   <source src="/resources/video/example-video1.webm" type='video/webm' />
    <source src="/resources/video/example-video1.mp4" type='video/mp4;'  />
</video>
```

### Describe the solution you'd like.

At the "Background Fallback" section there could be two options like there is for "Background Type". These could be "Image Icon" and "Video Icon" where when its clicked, it could have "Video Link" section with "File Type" selection. When the video is not played or supported by the browser it could play specified fallback video. Here is an ui example i quickly made in photoshop.

![ui-example](https://github.com/elementor/elementor/assets/60133120/1dbd7000-2cc8-4276-b0ae-a80b0715f4f0)

### Describe alternatives you've considered.

I've tried using the following script at the head of the pages that i use Background Videos. Idea was to duplicate the video element, and change the source of the duplicated element to be the fallback video which did not work.

```
<script src="https://code.jquery.com/jquery-3.6.4.min.js"></script>

<script>
$(document).ready(function() {
    function duplicateAndModifyVideos(selector, newSrc1, newSrc2) {
        var videosToDuplicate = $(selector + ' video');
        videosToDuplicate.each(function(index) {
            var newVideo = $(this).clone();

            newVideo.on('loadeddata', function() {
                newVideo[0].src = index === 0 ? newSrc1 : newSrc2;

                newVideo.off('loadeddata');
            });

            $(this).parent().append(newVideo);
        });
    }

    duplicateAndModifyVideos("#target-element-id1 .e-con-inner .elementor-background-video-container", "/resources/video/example-video2.mp4", "resources/video/example-video1.webm");
    duplicateAndModifyVideos("#target-element-id2 .e-con-inner .elementor-background-video-container", "/resources/video/example-video1.mp4", "resources/video/example-video2.webm");
});
</script>
```

### Additional context

Like i mentioned above, currently iOS devices seems like does not fully support WEBM videos and WEBM videos are great for performance. Changing the background video to MP4 from WEBM cost me a lot of points in performance insights. It is crucial especially if you use multiple videos and changing them to MP4 or simply not using them just for safari instead of using MP4 when it doesn't work seems to me a backwards way of including videos in your site.

Thank you.

### Agreement

- [X] I agree that my request may be closed without action if it doesn't meet all the requirements.

**Comment by alvarindev** [comment]
I think now is working fine webm, but in case you encounter like me problems with videos with alpha layer... I use this in my theme assets:

`
function updateVideoSource() {

  function isSafari() {
    const userAgent = navigator.userAgent;
    return (
      /^((?!chrome|android).)*safari/i.test(userAgent) &&
      /(iPhone|iPad|iPod|Macintosh)/i.test(userAgent)
    );
  }


  if (isSafari()) {
    console.log("es safari");
    const videoElements = document.getElementsByTagName("video");

    for (let i = 0; i < videoElements.length; i++) {
      const videoElement = videoElements[i];
      let currentSrc = videoElement.src;
      let newSrc = currentSrc.replace(/\.webm|\.mp4$/i, ".mov");
      console.log("src", newSrc);
      videoElement.src = newSrc;
    }
  }
}


document.addEventListener(
  "DOMContentLoaded",
  setTimeout(updateVideoSource, 500)
);

`

**Reply by xaelisart** [reply]
Yes, currently safari seems to be auto playing webm videos as well. Thank you for informing me on this. Still i believe this feature could be really useful for compatibility though.
